.github/workflows/static-code-analyzer.yml

```diff
@@ -0,0 +1,48 @@
+name: Static Code Analyzer
+on: [push, pull_request]
+
+jobs:
+  run-static-code-analyzer:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v2
+        id: checkout
+
+      - name: Install cppcheck
+        id: install-cppcheck
+        if: always()
+        run: |
+          sudo apt-get install cppcheck
+
+      - name: Clone engine repository
+        id: clone-engine-repo
+        if: always() && steps.install-cppcheck.outcome == 'success'
+        run: |
+          cd ..
+          rm -rf postgresql_modified_for_babelfish
+
+          if [[ $GITHUB_EVENT_NAME == "pull_request" ]]; then
+            ENGINE_BRANCH=$GITHUB_HEAD_REF
+          else
+            ENGINE_BRANCH=$GITHUB_REF_NAME
+          fi
+
+          $GITHUB_WORKSPACE/.github/scripts/clone_engine_repo "$GITHUB_REPOSITORY_OWNER" "$ENGINE_BRANCH"
```

Comment:
Can't we directly do this https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_3_X_DEV/.github/workflows/jdbc-tests.yml#L18?

Reply:
we don't need to build postgres. I think just cloning the repo would be sufficient

Reply:
Yeah that's true but composite action barely takes 30 sec due to ccache.

Reply:
We need to install other dependencies as well, if we want to build engine also. I don't think that there is need to build engine.